lib/puppet/provider/ios_interface/cisco_ios.rb

```diff
@@ -0,0 +1,124 @@
+      instance[:flowcontrol_receive] = 'off' unless instance[:flowcontrol_receive]
+      instance[:ip_dhcp_snooping_limit] = false unless instance[:ip_dhcp_snooping_limit]
+      instance = Puppet::Provider::IosInterface::CiscoIos.clean_logging_event(instance)
+      instance = Puppet::Provider::IosInterface::CiscoIos.boolean_link_status(instance)
+      cleaned_value << instance
+    end
+    cleaned_value
+  end
+
+  def self.commands_from_instance(property_hash, current)
+    commands = []
+    # If a value is current set in logging_event that is not in the applied manifest then
+    #   it needs to be removed.
+    if current[:logging_event]
+      to_remove = []
+      to_remove += if property_hash[:logging_event]
+                     current[:logging_event] - property_hash[:logging_event]
+                   else
+                     current[:logging_event]
+                   end
+      to_remove.each do |remove|
+        commands << "no #{PuppetX::CiscoIOS::Utility.build_commmands_from_attribute_set_values({ logging_event: remove }, commands_hash)[0]}"
+      end
+    end
+
+    # logging_event requires multiple commands to be built in most cases, as such it is handled
+    #   seperately from the remaining attributes
+    if property_hash[:logging_event]
+      property_hash[:logging_event].each do |event|
+        commands += PuppetX::CiscoIOS::Utility.build_commmands_from_attribute_set_values({ logging_event: event }, commands_hash)
+      end
+      property_hash.delete(:logging_event)
+    end
+
+    property_hash.each do |key, value|
+      property_hash[key] = Puppet::Provider::IosInterface::CiscoIos.false_to_unset(value)
+    end
+    commands += PuppetX::CiscoIOS::Utility.build_commmands_from_attribute_set_values(property_hash, commands_hash)
+    commands
+  end
+
+  # Returns 'unset' if the given calue is false
+  def self.false_to_unset(false_value)
+    return 'unset' if false_value == false
+    false_value
+  end
+
+  # Clears 'link-status' from 'logging-event'. This is done as it's default nature is
+  #   different than the remaining logging events.
+  def self.clean_logging_event(instance)
+    if instance[:logging_event]
+      if instance[:logging_event].class == Array
+        instance[:logging_event] -= ['link-status']
+      elsif instance[:logging_event].class == String
+        instance.delete(:logging_event) if instance[:logging_event] == 'link-status'
+      end
+    end
+    if instance[:logging_event] && instance[:logging_event].class == String
+      instance[:logging_event] = [instance[:logging_event]]
+    end
+    instance
+  end
+
+  # Converts 'logging_event_link_status' to a boolean value. The value only appears when
+  #   it is unset as it's default behaviour is to be set.
+  def self.boolean_link_status(instance)
+    instance[:logging_event_link_status] = if instance[:logging_event_link_status]
```

Comment:
Just trying to grasp the intent here, if the user supplies `true` you want it to be `false` for setting on the device? So does `false` for `logging_event_link_status` mean to enable that, and `true` means to disable it?

Reply:
This is only for retrieving the attribute, not setting it.
The default behaviour of this attribute is `true` and so it is only retrieved by `show running-config` when it is set to false.
So if the get value is found it means that the attribute is disabled and if it is not found it means that it is enabled.